examples/hello_world/client/android/java/com/google/oak/hello_world/MainActivity.java

```diff
@@ -22,59 +22,86 @@
 import android.widget.Button;
 import android.widget.EditText;
 import android.widget.TextView;
+import io.grpc.ManagedChannel;
+import io.grpc.okhttp.OkHttpChannelBuilder;
+import java.io.InputStream;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManagerFactory;
 
-import com.google.oak.hello_world.R;
-
-/*
- * Main class for the Oak Android "Hello, World" app.
- */
+/** Main activity for the Oak Android "Hello, World" app. */
 public class MainActivity extends Activity {
-  static {
-    // Load native library.
-    System.loadLibrary("client_app");
-  }
-
+  /** Handles initial setup on creation of the activity. */
   @Override
   public void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
-    
+
     Log.v("Oak", "Start application");
     setContentView(R.layout.activity_main);
-    
+
     Button helloButton = findViewById(R.id.helloButton);
-    helloButton.setOnClickListener(v -> onClick());
+    helloButton.setOnClickListener(v -> helloButtonOnClick());
 
     // Set default address.
     // Android emulator forwards `10.0.2.2` to the host machine.
     // https://developer.android.com/studio/run/emulator-networking
     EditText addressInput = findViewById(R.id.addressInput);
-    addressInput.setText("10.0.2.2:8888");
-  }
-
-  @Override
-  public void onDestroy() {
-    super.onDestroy();
+    addressInput.setText("10.0.2.2:8080");
   }
 
-  public void onClick() {
+  /** Handles click events from the helloButton. */
+  public void helloButtonOnClick() {
     EditText addressInput = findViewById(R.id.addressInput);
     String address = addressInput.getText().toString();
-
-    if (!address.equals(rpcAddress)) {
-      Log.v("Oak", "Create channel to: " + address);
-      createChannel(address);
-      rpcAddress = address;
-    }
-
     Log.v("Oak", "Say Hello");
     TextView helloTextView = findViewById(R.id.helloTextView);
-    String responce = sayHello("World");
-    Log.v("Oak", "Responce is: " + responce);
-    helloTextView.setText(responce);
+
+    // TODO(#988): Move blocking call ouf of UI thread.
+    String reply = sayHello(address, "World");
+    Log.v("Oak", "Response is: " + reply);
+    helloTextView.setText(reply);
   }
 
-  private native void createChannel(String address);
-  private native String sayHello(String name);
+  /** Calls the sayHello gRPC endpoint and returns the reply. */
+  private String sayHello(String address, String name) {
```

Comment:
Possibly use `Optional<String>` as the return type.

Reply:
Done.